cloud/services/groups/groups.go

```diff
@@ -86,6 +86,10 @@ func (s *Service) Delete(ctx context.Context) error {
 	defer span.End()
 
 	managed, err := s.IsGroupManaged(ctx)
+	if err != nil && azure.ResourceNotFound(err) {
```

Comment:
in #1118 we're not able to check the status of the RG at all because the sub ID is not found. The same issue would happen if the credentials had expired (GET fails). How does this change fix it? Is it because SubID not found returns a 404?

Reply:
yes, it was failing because the subscription name was wrong so the cluster never got created in Azure, but when you try to delete it, the controller was getting stuck on SubscriptionNotFound error in a loop